pymemcache/test/test_serde.py

```diff
@@ -1,24 +1,58 @@
 from unittest import TestCase
 
 from pymemcache.serde import (python_memcache_serializer,
-                              python_memcache_deserializer)
+                              python_memcache_deserializer, FLAG_PICKLE,
+                              FLAG_INTEGER, FLAG_LONG, FLAG_TEXT)
+import pytest
+import six
 
 
+class CustomInt(int):
+    """
+    Custom integer type for testing.
+
+    Entirely useless, but used to show that built in types get serialized and
+    deserialized back as the same type of object.
+    """
+    pass
+
+
+@pytest.mark.unit()
 class TestSerde(TestCase):
 
-    def check(self, value):
+    def check(self, value, expected_flags=0):
         serialized, flags = python_memcache_serializer(b'key', value)
+        assert flags == expected_flags
+
+        # pymemcache stores values as byte strings, so we immediately the value
+        # if needed so deserialized works as it would with a real server
+        if not isinstance(serialized, six.binary_type):
+            serialized = six.text_type(serialized).encode('ascii')
+
         deserialized = python_memcache_deserializer(b'key', serialized, flags)
         assert deserialized == value
 
-    def test_str(self):
-        self.check('value')
+    def test_bytes(self):
+        self.check(b'value')
```

Comment:
let's try this with unicode characters.
Also maybe explicitely ask for `FLAG_BYTES` as it should default to this?